fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/domain/SearchQueryRenderer.java

```diff
@@ -695,8 +695,8 @@ private void fetchCommonTokenValues(Set<Long> ctvs, String searchValue) throws F
     private void addCommonTokenValueIdFilter(WhereFragment where, String paramAlias, Collection<Long> ctvs) throws FHIRPersistenceException {
         final List<Long> ctvList = new ArrayList<>(ctvs);
         if (ctvList.isEmpty()) {
-            // use -1...resulting in no data
-            where.col(paramAlias, COMMON_TOKEN_VALUE_ID).eq(-1L);
+            // the column is NOT NULL...resulting in no data
+            where.col(paramAlias, COMMON_TOKEN_VALUE_ID).isNull();
```

Comment:
I'm under the impression that -1 is more selective than isNull ... 
Curious as to how the dbms treats is null versus = 0,1  I thought with Null it maintained an extra bit of data in the dbms. (at least on db2)

Reply:
For our optimization you want to be using
```
where.col(paramAlias, COMMON_TOKEN_VALUE_ID).eq().literalNull();
```

or something like that. Although the optimizer might be able to see that the column is not null, and therefore could short-circuit the access, I would think that foo_id = NULL is still a better bet - unless you've seen something different in the query plans.

Reply:
> I'm under the impression that -1 is more selective than isNull ...

Is that true for a column that is marked as NOT NULL?

> Although the optimizer might be able to see that the column is not null, and therefore could short-circuit the access, I would think that foo_id = NULL is still a better bet - unless you've seen something different in the query plans.

As I mentioned in the description of the PR, ` = NULL` was throwing SQLSyntaxException (at least while using derby). Here was the full SQL and the error it produced:
```
Aug 23, 2021 8:01:33 AM com.ibm.fhir.persistence.jdbc.dao.impl.FHIRDbDAOImpl severe
SEVERE: 
      SELECT COUNT(*) AS CNT 
        FROM Basic_LOGICAL_RESOURCES AS LR0
       WHERE LR0.IS_DELETED = 'N'
         AND EXISTS (
      SELECT 1 
        FROM Basic_LOGICAL_RESOURCES AS LR1 
  INNER JOIN Basic_RESOURCE_TOKEN_REFS AS P2 ON P2.LOGICAL_RESOURCE_ID = LR1.LOGICAL_RESOURCE_ID
         AND P2.PARAMETER_NAME_ID = 20013
         AND (P2.COMMON_TOKEN_VALUE_ID = NULL) 
  INNER JOIN Basic_STR_VALUES AS P3 ON P3.LOGICAL_RESOURCE_ID = LR1.LOGICAL_RESOURCE_ID
         AND P3.PARAMETER_NAME_ID = 20018
         AND (P3.STR_VALUE_LCASE LIKE ? ESCAPE '+')
       WHERE LR1.LOGICAL_RESOURCE_ID = LR0.LOGICAL_RESOURCE_ID) [probeId=7f-0-0-1-92233e01-fcf2-40ea-8906-b35ae9e782be]
java.sql.SQLSyntaxErrorException: Syntax error: Encountered "NULL" at line 10, column 42.
	at org.apache.derby.impl.jdbc.SQLExceptionFactory.getSQLException(Unknown Source)
	at org.apache.derby.impl.jdbc.Util.generateCsSQLException(Unknown Source)
	at org.apache.derby.impl.jdbc.TransactionResourceImpl.wrapInSQLException(Unknown Source)
	at org.apache.derby.impl.jdbc.TransactionResourceImpl.handleException(Unknown Source)
	at org.apache.derby.impl.jdbc.EmbedConnection.handleException(Unknown Source)
	at org.apache.derby.impl.jdbc.ConnectionChild.handleException(Unknown Source)
	at org.apache.derby.impl.jdbc.EmbedPreparedStatement.<init>(Unknown Source)
	at org.apache.derby.impl.jdbc.EmbedPreparedStatement42.<init>(Unknown Source)
	at org.apache.derby.jdbc.Driver42.newEmbedPreparedStatement(Unknown Source)
	at org.apache.derby.impl.jdbc.EmbedConnection.prepareStatement(Unknown Source)
	at org.apache.derby.impl.jdbc.EmbedConnection.prepareStatement(Unknown Source)
	at com.ibm.fhir.database.utils.pool.PooledConnection.prepareStatement(PooledConnection.java:429)
	at com.ibm.fhir.database.utils.query.QueryUtil.prepareSelect(QueryUtil.java:49)
	at com.ibm.fhir.persistence.jdbc.dao.impl.FHIRDbDAOImpl.runCountQuery(FHIRDbDAOImpl.java:320)
	at com.ibm.fhir.persistence.jdbc.dao.impl.ResourceDAOImpl.searchCount(ResourceDAOImpl.java:783)
	at com.ibm.fhir.persistence.jdbc.impl.FHIRPersistenceJDBCImpl.newSearch(FHIRPersistenceJDBCImpl.java:715)
	at com.ibm.fhir.persistence.jdbc.impl.FHIRPersistenceJDBCImpl.search(FHIRPersistenceJDBCImpl.java:676)
	at com.ibm.fhir.persistence.test.common.AbstractPersistenceTest.executeCompartmentQuery(AbstractPersistenceTest.java:237)
	at com.ibm.fhir.persistence.test.common.AbstractPersistenceTest.runCompartmentQueryTest(AbstractPersistenceTest.java:218)
	at com.ibm.fhir.persistence.search.test.AbstractSearchCompartmentTest.compartmentSearchReturnsResource(AbstractSearchCompartmentTest.java:207)
	at com.ibm.fhir.persistence.search.test.AbstractSearchCompartmentTest.assertCompartmentSearchDoesntReturnSavedResource(AbstractSearchCompartmentTest.java:186)
	at com.ibm.fhir.persistence.search.test.AbstractSearchCompartmentTest.testSearchPatientCompartment_relativeReference_string(AbstractSearchCompartmentTest.java:99)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:132)
	at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:599)
	at org.testng.internal.TestInvoker.invokeTestMethod(TestInvoker.java:174)
	at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:46)
	at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:822)
	at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:147)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
	at java.util.ArrayList.forEach(ArrayList.java:1257)
	at org.testng.TestRunner.privateRun(TestRunner.java:764)
	at org.testng.TestRunner.run(TestRunner.java:585)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:384)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:378)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:337)
	at org.testng.SuiteRunner.run(SuiteRunner.java:286)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:96)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1218)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1140)
	at org.testng.TestNG.runSuites(TestNG.java:1069)
	at org.testng.TestNG.run(TestNG.java:1037)
	at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
	at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:251)
	at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:77)
Caused by: ERROR 42X01: Syntax error: Encountered "NULL" at line 10, column 42.
	at org.apache.derby.iapi.error.StandardException.newException(Unknown Source)
	at org.apache.derby.iapi.error.StandardException.newException(Unknown Source)
	at org.apache.derby.impl.sql.compile.ParserImpl.parseStatementOrSearchCondition(Unknown Source)
	at org.apache.derby.impl.sql.compile.ParserImpl.parseStatement(Unknown Source)
	at org.apache.derby.impl.sql.GenericStatement.prepMinion(Unknown Source)
	at org.apache.derby.impl.sql.GenericStatement.prepare(Unknown Source)
	at org.apache.derby.impl.sql.conn.GenericLanguageConnectionContext.prepareInternalStatement(Unknown Source)
	... 44 more
```

I don't think this one is that important, so if `IS NULL` is not what we want then maybe we should just leave it as `= -1`.